napari/utils/theme.py

```diff
@@ -13,64 +15,95 @@
     use_gradients = False
 
 from ..utils.translations import trans
-
-
-def __getattr__(attr):
-    if attr == "palettes":
-        warnings.warn(
-            trans._(
-                "palette is deprecated and will be removed after version 0.4.6. Please use get_theme and register_theme instead",
-                deferred=True,
-            ),
-            category=FutureWarning,
-            stacklevel=2,
-        )
-        return _themes
-    raise AttributeError
-
-
-_themes = {
-    'dark': {
-        'folder': 'dark',
-        'background': 'rgb(38, 41, 48)',
-        'foreground': 'rgb(65, 72, 81)',
-        'primary': 'rgb(90, 98, 108)',
-        'secondary': 'rgb(134, 142, 147)',
-        'highlight': 'rgb(106, 115, 128)',
-        'text': 'rgb(240, 241, 242)',
-        'icon': 'rgb(209, 210, 212)',
-        'warning': 'rgb(153, 18, 31)',
-        'current': 'rgb(0, 122, 204)',
-        'syntax_style': 'native',
-        'console': 'rgb(0, 0, 0)',
-        'canvas': 'black',
-    },
-    'light': {
-        'folder': 'light',
-        'background': 'rgb(239, 235, 233)',
-        'foreground': 'rgb(214, 208, 206)',
-        'primary': 'rgb(188, 184, 181)',
-        'secondary': 'rgb(150, 146, 144)',
-        'highlight': 'rgb(163, 158, 156)',
-        'text': 'rgb(59, 58, 57)',
-        'icon': 'rgb(107, 105, 103)',
-        'warning': 'rgb(255, 18, 31)',
-        'current': 'rgb(253, 240, 148)',
-        'syntax_style': 'default',
-        'console': 'rgb(255, 255, 255)',
-        'canvas': 'white',
+from .events import EventedModel
+from .events.containers._evented_dict import EventedDict
+
+THEME_KEYS = [
+    "folder",
+    "background",
+    "foreground",
+    "primary",
+    "highlight",
+    "text",
+    "icon",
+    "warning",
+    "current",
+    "syntax_style",
+    "console",
+    "canvas",
+]
+
+
+class Theme(EventedModel):
+    """Theme model."""
+
+    folder: str
+    syntax_style: str
```

Comment:
I've added a value validator that will check against `pygments.styles.STYLE_MAP`. I don't think `pygments` is a direct dependency so this might have to optional?

Reply:
> I don't think pygments is a direct dependency so this might have to optional?

That's true, but the only reason we have this option in the theme is basically to support `napari-console`, which depends on qtconsole, and _that_ depends on `pygments` (and if it stopped depending on pygments, then we'd basically need to come up with new themeing anyway).  So, I'd say we should either remove this from our theme entirely (and leave it to `napari-console` to figure out how to theme itself)... or we should just use pygments for now and leave that decision for when we don't require `napari-console`.  If we keep it, here's one option for validating against a set of string literals:

```py
from typing_extensions import Literal as L
import pygments

class Theme(EventedModel):
    s: Union[tuple(L[x] for x in pygments.styles.STYLE_MAP)]
```

Reply:
> or we should just use pygments for now and leave that decision for when we don't require napari-console

Maybe we do this for now and then refactor when we pull out napari-console? Just seems a little easier, but we can also deal with now

Reply:
Agree, I am happy to change this when napari-console is no longer a dependency in the future.